File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
  def isValidSudoku(self, board: List[List[str]]) -> bool:
    s = set()
    for row in range(9):
      for col in range(9):
        if board[row][col] != '.':
          val = int(board[row][col])
          box = row // 3 * 3 + col // 3
          if row+1<<4|val in s or col+1<<8|val in s or box+1<<12|val in s:
            return False
          s.add(row+1<<4|val)
          s.add(col+1<<8|val)
          s.add(box+1<<12|val)
    return True
        

File: 0036-valid-sudoku/0036-valid-sudoku.php
use Ds\Set as Set;

class Solution {

  /**
    * @param String[][] $board
    * @return Boolean
    */
  function isValidSudoku($board) {
    $set = [];
    for ($row = 0; $row < count($board); $row++) {
      for ($col = 0; $col < count($board); $col++) {
        $val = (int)$board[$row][$col];
        if ($val) {
          $box = floor($row / 3) * 3 + floor($col / 3);
          if (
            $set[$row + 1 << 4 | $val] ||
            $set[$col + 1 << 8 | $val] ||
            $set[$box + 1 << 12 | $val]
          ) return false;
          $set[$row + 1 << 4 | $val] = true;
          $set[$col + 1 << 8 | $val] = true;
          $set[$box + 1 << 12 | $val] = true;
        }
      } 
    } 
    return true;
  }
}